src/views/design/ScriptDetail/index.tsx

```diff
@@ -300,9 +304,11 @@ const ScriptDetail = withStyles(styles)(
                                 onChange={(e) => this.updateScript({ name: e.target.value })}
                                 required={this.isCreateScriptRoute()}
                                 InputProps={{
-                                    readOnly: !this.isCreateScriptRoute(),
+                                    readOnly: !this.isCreateScriptRoute()
+                                    || !checkAuthority(SECURITY_PRIVILEGES.S_SCRIPTS_WRITE, 'PUBLIC'),
```

Comment:
Can we already use the security group name of the script in question instead of the fixed value 'PUBLIC'.